fs.go

```diff
@@ -726,7 +736,12 @@ func (h *fsHandler) handleRequest(ctx *RequestCtx) {
 			}
 		} else if err != nil {
 			ctx.Logger().Printf("cannot open file %q: %s", filePath, err)
-			ctx.Error("Cannot open requested path", StatusNotFound)
+			if h.pathNotFound == nil {
+				ctx.Error("Cannot open requested path", StatusNotFound)
+			} else {
+				h.pathNotFound(ctx)
+				ctx.SetStatusCode(StatusNotFound)
```

Comment:
pathNotFound may be used to serve SPA index page, so we should not overwrite status code if it was already set